### ISSUE TITLE
ElasticSearch update-advanced-visibility-schema job keep detecting schema version 3 instead of recent one

### ISSUE THREAD

**Issue by Tassatux** [alexandrevilain/temporal-operator#462]
When I upgraded my cluster from 1.20.3 to 1.21.4, I've got an error on `update-advanced-visibility-schema` job:

```
Expected schema version: v5
Current schema version: v3
Schema version upgrade is needed
Can't do Elasticsearch schema upgrade for no-following version numbers. (eg. from v1 to v2, but not from v1 to v3)
```

I've tried to manually upgrade to schema v4 using steps from `update-advanced-visibility-schema.sh` script and let the job re-run, but it failed at the same step.

It look like the [condition to detect schema v3](https://github.com/alexandrevilain/temporal-operator/blob/v0.15.0/internal/resource/persistence/template.go#L210) is still "true" on newer version and as the script consider it found the current version, it ignore newer version detection.

To finish the migration I had to scale to 0 the temporal-operator deployment, edit the ConfigMap which contain the `update-advanced-visibility-schema.sh` script and change the `is_v3` condition to make it false. Once the job ran successfully, I could scale up temporal-operator and let it finish the upgrade.

This also happened to me when deleting, then re-deploying Temporal cluster, the schema was already upgraded in v5 but I still had the same issue.

temporal-operator version: 0.15.0

**Comment by alexandrevilain** [comment]
Hi @Tassatux ! 
Thanks for the issue, let me dig inside it and I'll come back to you :)

**Comment by alexandrevilain** [comment]
Thanks for the report! I did the fix, it will be released in 0.15.1 ! 
Now when upgrading from 1.20.3 to 1.21.4, the job prints the following lines:

```
Expected schema version: v5
Current schema version: v4
Schema version upgrade is needed
Upgrading to schema v5
{
  "acknowledged": true
}
true
```

Thanks for the bug report's quality @Tassatux !
